### PR TITLE
specs: add predeploys

### PR DESCRIPTION
**Pull request by tynes**
**Description**
Add predeploys spec. These need to be upgraded in a backwards compatible way

Replaces https://github.com/ethereum-optimism/optimistic-specs/pull/379

### CODE REVIEW

**Review by mslipper — APPROVED** [review]

**Review by trianglesphere — COMMENTED** [review]
I cross referenced this list with the contracts (after https://github.com/ethereum-optimism/optimism/pull/2580) and I think I'm only seeing one missing contract in the other PR.

One other thing that got dropped is that we need to pull these addresses into `go` and make sure that we use them in the right places in our test systems (we are currently using self consistently wrong addresses which is why it works now). - this is not a blocker for this PR.

File: specs/predeploys.md [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+| Name                         | Address                                    | Introduced | Deprecated |
+| ---------------------------- | ------------------------------------------ | ---------- | ---------- |
+| OVM\_L2ToL1MessagePasser     | 0x4200000000000000000000000000000000000000 | Legacy     | No         |
+| OVM\_DeployerWhitelist       | 0x4200000000000000000000000000000000000002 | Legacy     | Yes        |
+| OVM\_ETH                     | 0xDeadDeAddeAddEAddeadDEaDDEAdDeaDDeAD0000 | Legacy     | Yes        |
+| WETH9                        | 0x4200000000000000000000000000000000000006 | Legacy     | No         |
+| L2CrossDomainMessenger       | 0x4200000000000000000000000000000000000007 | Legacy     | No         |
+| L2StandardBridge             | 0x4200000000000000000000000000000000000010 | Legacy     | No         |
+| SequencerFeeVault            | 0x4200000000000000000000000000000000000011 | Legacy     | No         |
+| OptimismMintableTokenFactory | 0x4200000000000000000000000000000000000012 | Legacy     | No         |
```
[the remark this answers is not in the document]
Comment (trianglesphere) [reply specs/predeploys.md:49]: Edit, I found the `OptimismMintableTokenFactory` contract in the `universal` folder

**Comment by trianglesphere** [comment]
@tynes are you planning to push this forwards?

**Comment by tynes** [comment]
I'd also like to remove all references to the interfaces in this eventually and instead link to an `interfaces` directory in `contracts-bedrock` so that there is a single source of truth

**Review by trianglesphere — APPROVED** [review]
lgtm

I'm not sure where we should record the follow ups (ensure L1 Block is the contract name & reminder to upgrade to interfaces).

**Pull request merged**